pkg/messagebroker/kafka.go

```diff
@@ -571,3 +572,66 @@ func (k *KafkaBroker) Close(ctx context.Context) error {
 
 	return nil
 }
+
+// AddTopicPartitions adds partitions to an existing topic
+func (k *KafkaBroker) AddTopicPartitions(ctx context.Context, request AddTopicPartitionRequest) (*AddTopicPartitionResponse, error) {
+	messageBrokerOperationCount.WithLabelValues(env, Kafka, "AddTopicPartitions").Inc()
+
+	startTime := time.Now()
+	defer func() {
+		messageBrokerOperationTimeTaken.WithLabelValues(env, Kafka, "AddTopicPartitions").Observe(time.Now().Sub(startTime).Seconds())
+	}()
+
+	tp := NormalizeTopicName(request.Name)
+	metadata, merr := k.Admin.GetMetadata(&tp, false, 1000)
+	if merr != nil {
+		logger.Ctx(ctx).Errorw("kafka: admin getMetadata() failed", "topic", request.NumPartitions, "error", merr.Error())
+		messageBrokerOperationError.WithLabelValues(env, Kafka, "AddTopicPartitions", merr.Error()).Inc()
+		return nil, merr
+	}
+
+	// fetch the topic metadata and check the current partition count
+	if _, ok := metadata.Topics[tp]; !ok {
+		// invalid topic
+		return nil, fmt.Errorf("topic [%v] not found", tp)
+	}
+
+	// metadata contains non-existent topic names as well with partition count as zero
+	currPartitionCount := len(metadata.Topics[tp].Partitions)
+	if currPartitionCount == 0 {
+		// invalid topic
+		return nil, fmt.Errorf("topic [%v] not found", tp)
+	}
+
+	if request.NumPartitions == currPartitionCount {
```

Comment:
this validation should be done via topic registry layer ?

Reply:
Since `messagebroker` is a `pkg` this function can ideally be called from anywhere without prior topic registry lookups, having this **check** here completes the function on its own. Similarly how we validate whether the topic name sent was valid or not.
Will change if this doesn't sound correct. Thoughts?